### PR TITLE
Add alwaysShowTimestamps and others to RoomView setting watchers

### PR DESCRIPTION
**Pull request by robintown**
to allow them to update on the fly. This also modifies the setting watchers to avoid an unnecessary settings lookup.

Based on #6243 to keep ahead of conflicts.
Closes vector-im/element-web#17748.

### CODE REVIEW

**Review by turt2live — APPROVED** [review]
@t3chguy what's blocking this?

**Comment by t3chguy** [comment]
Based on another PR which hasn't landed yet

**Comment by turt2live** [comment]
ah, alright then. Didn't even realize it had a strange base applied 😅

**Comment by t3chguy** [comment]
image


One of my extensions is a lifesaver

**Comment by jryans** [comment]
@robintown Hmm, I wonder if we could get a version of this targetting current `develop`?

https://github.com/matrix-org/matrix-react-sdk/pull/6243 is large enough that it sound wait until after tomorrow's RC, while this fixes what appears to be a regression in the current release, so ideally we can merge it now to ship the fix sooner.

**Comment by t3chguy** [comment]
I don't think it's a regression. You always had to change rooms for certain settings to get picked up

**Comment by jryans** [comment]
Ah okay, carry on then. 😅

**Comment by robintown** [comment]
@t3chguy Could you please reopen and change base to develop? (somehow I thought GitHub would handle things automatically)

**Comment by t3chguy** [comment]
Very weird, when I stack my own PRs it just does the right thing, maybe it doesn't over fork boundaries for whatever reason

**Comment by t3chguy** [comment]
@turt2live are you happy for this to land?

**Comment by turt2live** [comment]
> Very weird, when I stack my own PRs it just does the right thing, maybe it doesn't over fork boundaries for whatever reason

The branch deletion was what did the wrong thing, ftr.

> @turt2live are you happy for this to land?

I believe so, but will sanity check

**Comment by t3chguy** [comment]
I always delete my branches, heck a browser extension I use automatically does it

File: src/components/structures/RoomView.tsx [hunk 1]
```diff
@@ -263,14 +271,26 @@ export default class RoomView extends React.Component<IProps, IState> {
...
+                this.setState({ showTwelveHourTimestamps: value as boolean }),
+            ),
+            SettingsStore.watchSetting("readMarkerInViewThresholdMs", null, (...[,,, value]) =>
+                this.setState({ readMarkerInViewThresholdMs: value as number }),
+            ),
+            SettingsStore.watchSetting("readMarkerOutOfViewThresholdMs", null, (...[,,, value]) =>
+                this.setState({ readMarkerOutOfViewThresholdMs: value as number }),
+            ),
+            SettingsStore.watchSetting("showHiddenEventsInTimeline", null, (...[,,, value]) =>
+                this.setState({ showHiddenEventsInTimeline: value as boolean }),
```
Comment (turt2live) [inline src/components/structures/RoomView.tsx:293]: I'm going to be so happy when the new settings structure lands and we don't have to deal with `as crap` anymore.

**Review by turt2live — APPROVED** [review]
:shipit:

**Pull request merged**